src/mango.ts

```diff
@@ -0,0 +1,493 @@
+  mangoGroup: Group,
+  perpMarket: PerpMarket,
+  spotMarket: Market,
+  fillFeed: WebSocket,
+  deltaHedgeCount: number,
+): Promise<void> {
+  await mangoAccount.reload(mangoClient);
+  await mangoGroup.reloadAll(mangoClient);
+  // Cleanup from previous runs.
+  await cancelStaleMangoOrders(scalper, mangoClient, mangoAccount, mangoGroup, perpMarket);
+
+  // Avoid unsafe recursion.
+  if (deltaHedgeCount > MAX_DELTA_HEDGES) {
+    console.log(scalper.symbol, 'Max Hedges exceeded without getting to neutral');
+    return;
+  }
+
+  // Calc DIP delta for new position
+  const fairValue = perpMarket.uiPrice;
+  const dipTotalDelta = getDIPDelta(dipProduct, fairValue, scalper.symbol);
+
+  // Get Mango delta position
+  const mangoPerpDelta = mangoAccount.getPerpPositionUi(mangoGroup, perpMarket.perpMarketIndex);
+  let mangoSpotDelta = 0;
+  try {
+    mangoSpotDelta = mangoAccount.getTokenBalanceUi(
+      mangoGroup.getFirstBankByMint(tokenToSplMint(scalper.symbol)),
+    );
+  } catch (err) {
+    console.log(scalper.symbol, 'No Mango Token Balance Possible');
+  }
+
+  // Get all spot positions Option Vault, Risk Manager, Mango Tester
+  const spotDelta = await getSpotDelta(
+    scalper.connection,
+    scalper.symbol,
+    scalper.owner,
+    spotMarket,
+  );
+
+  // Get Total Delta Position to hedge
+  let hedgeDeltaTotal = IS_DEV
+    ? 0.1
+    : mangoPerpDelta + dipTotalDelta + spotDelta + mangoSpotDelta + scalper.deltaOffset;
+
+  // Check whether we need to hedge.
+  const dipTotalGamma = getDIPGamma(dipProduct, fairValue, scalper.symbol);
+  const stdDevSpread = (scalper.impliedVol / Math.sqrt(SEC_PER_YEAR
+    / SCALPER_WINDOW_SEC)) * scalper.zScore;
+  const deltaThreshold = Math.max(
+    dipTotalGamma * stdDevSpread * fairValue * GAMMA_THRESHOLD,
+    scalper.minSize,
+  );
+  if (Math.abs(hedgeDeltaTotal) < (deltaThreshold)) {
+    console.log(scalper.symbol, 'Delta Netural', hedgeDeltaTotal, '<', deltaThreshold);
+    return;
+  }
+
+  // TODO: Add slippage delta adjustment
+  const slippageTolerance = Math.min(stdDevSpread / 2, slippageMax.get(scalper.symbol));
+  const IS_BUYSIDE = hedgeDeltaTotal < 0;
+  let hedgePrice = IS_BUYSIDE
+    ? fairValue * (1 + slippageTolerance)
+    : fairValue * (1 - slippageTolerance);
+  const slippageDIPDelta = getDIPDelta(dipProduct, hedgePrice, scalper.symbol);
+  const dipDeltaDiff = slippageDIPDelta - dipTotalDelta;
+  hedgeDeltaTotal += dipDeltaDiff;
+  console.log(scalper.symbol, 'Adjust Slippage Delta by', dipDeltaDiff, 'to', -hedgeDeltaTotal);
+  const hedgeSide = IS_BUYSIDE ? HedgeSide.buy : HedgeSide.sell;
+
+  const notionalThreshold = deltaThreshold * fairValue;
+  const notionalAmount = -hedgeDeltaTotal * fairValue;
+  if ((notionalAmount < notionalThreshold && hedgeSide === HedgeSide.buy)
+       || (notionalAmount > notionalThreshold && hedgeSide === HedgeSide.sell)) {
+    console.log(scalper.symbol, 'Delta Netural: Slippage', deltaThreshold);
```

Comment:
Log line should say both sides of what you are comparing. I think as is, should it be slippage limit, not slippage.

Reply:
Same as above. Condensed